pennylane/workflow/qnode.py

```diff
@@ -526,7 +526,7 @@ def __init__(
         self.max_expansion = max_expansion
         cache = (max_diff > 1) if cache == "auto" else cache
 
-        if mcm_method not in ("deferred", "one-shot", None):
+        if mcm_method not in ("deferred", "one-shot", "tree-traversal", None):

```

Comment:
There is an error in `QNode._execution_component` if a user requests `one-shot` without finite shots. The check for that error should now also include `tree-traversal`.

Reply:
Thanks for pointing this out.